From eb4d1be6a9cd7b982515a204bd8ff66059c98155 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Thu, 4 Jul 2024 08:52:27 -0700
Subject: [PATCH] tools/testing/selftests/kselftest/runner: warn non-executable
 tests only when really cannot execute

If a test program is not having the executable permission, runner.sh
warns the fact, and then checks if it has a shebang.  If the shebang is
not found, it simply fails.  Otherwise, it parses the interpreter and
execute the test with it.  The warning can be helpful in the former case
since it allows users know why it failed.  But in the latter case, it is
just a noise that users usually don't really care.  Print the message
only for the failing case.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/kselftest/runner.sh | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tools/testing/selftests/kselftest/runner.sh b/tools/testing/selftests/kselftest/runner.sh
index c0070ef649b9..d266460309e2 100644
--- a/tools/testing/selftests/kselftest/runner.sh
+++ b/tools/testing/selftests/kselftest/runner.sh
@@ -112,13 +112,12 @@ run_one()
 		eval kselftest_cmd_args="\$${kselftest_cmd_args_ref:-}"
 		cmd="$stdbuf ./$BASENAME_TEST $kselftest_cmd_args"
 		if [ ! -x "$TEST" ]; then
-			echo "# Warning: file $TEST is not executable"
-
 			if [ $(head -n 1 "$TEST" | cut -c -2) = "#!" ]
 			then
 				interpreter=$(head -n 1 "$TEST" | cut -c 3-)
 				cmd="$stdbuf $interpreter ./$BASENAME_TEST"
 			else
+				echo "# file $TEST is not executable"
 				echo "not ok $test_num $TEST_HDR_MSG"
 				return
 			fi
-- 
2.39.2

